server/checks/PullRequest.js

```diff
@@ -0,0 +1,99 @@
+import Check from './Check'
+import { logger } from '../../common/debug'
+import * as EVENTS from '../model/GithubEvents'
+
+const CHECK_TYPE = 'pullrequest'
+const ACTIONS = ['opened', 'edited', 'reopened', 'synchronize']
+
+const ISSUE_PATTERN = /^[^#\s]*#\d+$/
+// Grubber's pattern
+const URL_PATTERN = /\b((?:[a-z][\w-]+:(?:\/{1,3}|[a-z0-9%])|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}\/)(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'".,<>?«»“”‘’]))/i
+
+const info = logger(CHECK_TYPE, 'info')
+const error = logger(CHECK_TYPE, 'error')
+
+const status = (description, state = 'success') => ({
+  description,
+  state,
+  context: 'zappr/check/pull-request'
+})
+
+const isLongEnough = (str, requiredLength) => (str || '').length > requiredLength
+const containsPattern = pattern => str => (str || '').split(' ')
+  .findIndex(s => pattern.test(s)) > -1
+const containsUrl = containsPattern(URL_PATTERN)
+const containsIssueNumber = containsPattern(ISSUE_PATTERN)
+
+export default class PullRequest extends Check {
+  static TYPE = CHECK_TYPE
+  static NAME = 'Pull request check'
+  static HOOK_EVENTS = [EVENTS.PULL_REQUEST]
+
+  /**
+   * @param {GithubService} github
+   */
+  constructor(github) {
+    super() // TODO: create constructor that receives `github`
+    this.github = github
+  }
+
+  async execute(config, hookPayload, token) {
+    const { action, pull_request: pr, repository: repo } = hookPayload
+
+    if (ACTIONS.indexOf(action) === -1 || !pr || 'open' !== pr.state) {
+      info(`${repo.full_name}#${pr.number}: Nothing to do, action was "${action}" with state "${pr.state}".`)
+      return
+    }
+
+    await this.validate(config, pr, repo, token)
+  }
+
+  /**
+   * Should do next validation steps:
+   *  * check if title's length is more than required length
+   *  * check if body has one of the following (in order):
+   *    * at least one issue number
+   *    * at least one link
+   *    * it's length is more than required length
+   *
+   * @param config config object
+   * @param pr Github's PR object
+   * @param repo Github's repository object
+   * @param token access token
+   */
+  async validate(config, pr, repo, token) {
+    const { head: { sha } } = pr
+    const { owner: { login: user } } = repo
+    const { pullrequest: {
+      title: { length: titleLength },
+      body: {
+        length: bodyLength,
+        checks: { url: shouldCheckUrl, issue: shouldCheckIssue }
```

Comment:
What happens when there is a `null` field along the way (ie `pullrequest.body` is in the config, but not `pullrequest.body.checks`? Maybe it's better to use `getIn` from our utils for safe access of nested props.